stablehlo/dialect/VhloOps.td

```diff
@@ -762,213 +880,181 @@ def VHLO_SelectAndScatterOpV1 : VHLO_Op<"select_and_scatter", "0.3.0", "current"
-    OptionalAttr<VHLO_AnyAttr>:$indices_are_sorted
+    VHLO_AnyType:$a,
+    VHLO_AnyType:$b,
+    VHLO_AnyAttr:$left_side,
+    VHLO_AnyAttr:$lower,
+    VHLO_AnyAttr:$unit_diagonal,
+    VHLO_AnyAttr:$transpose_a
   );
-  let results = (outs VHLO_AnyType);
+  let results = (outs VHLO_AnyType:$result);
 }
 
-def VHLO_DynamicConvOpV1 : VHLO_Op<"dynamic_conv", "0.3.0", "current"> {
-  let arguments = !con(
-    (ins
-       VHLO_AnyType:$lhs,
-       VHLO_AnyType:$rhs,
-       VHLO_AnyType:$d_padding),
-    VHLO_ConvolutionAttributesV1.attributes);
-  let results = (outs VHLO_AnyType);
+def VHLO_TupleOpV1 : VHLO_Op<"tuple", "0.3.0", "current"> {
+  let arguments = (ins Variadic<VHLO_AnyType>:$val);
+  let results = (outs VHLO_AnyType:$result);
 }
 
-def VHLO_ComputeReshapeShapeOpV1 : VHLO_Op<"compute_reshape_shape", "0.3.0", "current"> {
-  let arguments = (ins VHLO_AnyType:$num_elements, VHLO_AnyType:$dynamic_shape);
+// TODO(#3): UnaryEinsumOp is not part of the StableHLO spec.
+// This operation is on its way out of StableHLO, so it is not included in
+// the StableHLO specification.
+def VHLO_UnaryEinsumOpV1 : VHLO_Op<"unary_einsum", "0.3.0", "current"> {
+  let arguments = (ins
+    VHLO_AnyType:$operand,
+    VHLO_AnyAttr:$einsum_config
+  );
   let results = (outs VHLO_AnyType:$result);
 }
 
-def VHLO_CstrReshapableOpV1 : VHLO_Op<"cstr_reshapable", "0.3.0", "current"> {
+// TODO(#588): UniformDequantizeOp is not part of the StableHLO spec.
+// This operation is a work in progress, so it is not yet included in
+// the StableHLO specification.
+def VHLO_UniformDequantizeOpV1 : VHLO_Op<"uniform_dequantize", "0.3.0", "current"> {
+  let arguments = (ins VHLO_AnyType:$operand);
   let results = (outs VHLO_AnyType:$result);
-  let arguments = (ins VHLO_AnyType:$num_elements, VHLO_AnyType:$dynamic_shape);
 }
 
-def VHLO_FuncOpV1 : VHLO_Op<"func", "0.3.0", "current"> {
-  let arguments = (ins VHLO_AnyAttr:$sym_name,
-                       VHLO_AnyAttr:$function_type,
-                       OptionalAttr<VHLO_AnyAttr>:$sym_visibility,
-                       OptionalAttr<VHLO_AnyAttr>:$arg_attrs,
-                       OptionalAttr<VHLO_AnyAttr>:$res_attrs);
-  let regions = (region VHLO_AnyRegion:$body);
-  let assemblyFormat = "custom<FunctionBody>($sym_name, $body, $function_type) attr-dict";
+// TODO(#588): UniformQuantizeOp is not part of the StableHLO spec.
+// This operation is a work in progress, so it is not yet included in
+// the StableHLO specification.
+def VHLO_UniformQuantizeOpV1 : VHLO_Op<"uniform_quantize", "0.3.0", "current"> {
+  let arguments = (ins VHLO_AnyType:$operand);
+  let results = (outs VHLO_AnyType:$result);
 }
 
-def VHLO_CallOpV1 : VHLO_Op<"call", "0.3.0", "current"> {
-  let arguments = (ins VHLO_AnyAttr:$callee, Variadic<VHLO_AnyType>:$operands);
-  let results = (outs Variadic<VHLO_AnyType>);
+def VHLO_WhileOpV1 : VHLO_Op<"while", "0.3.0", "current"> {
+  let arguments = (ins Variadic<VHLO_AnyType>:$operand);
```

Comment:
Is there a TODO on this to get the name more standardized? We use `inputs` `operands` and now `operand` to describe variadic input arguments, a bit strange.

Similar question with results vs outputs

Reply:
Yes, #351.